src/utils/number.spec.ts

```diff
@@ -111,6 +111,15 @@ describe("BigDecimal", () => {
     expect(negativeZero).toBe("-0");
   });
 
+  it("correctly formats long decimal numbers", () => {
+    numberStringFormatter.numberFormatOptions = {
+      locale: "en",
+      numberingSystem: "latn",
+      useGrouping: true
+    };
+    expect(new BigDecimal("123.123456789").format(numberStringFormatter)).toBe("123.123456789");
```

Comment:
can we add one or two other "expects" here to confirm this change for other locales and non latin numbering systems?

Reply:
I added coverage for all of the supported numbering systems in the locale spec tests. `numberStringFormatter.localize` uses `BigDecimal.format`.